sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayAsyncClient.java

```diff
@@ -64,34 +98,53 @@
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationUserIdentifier communicationUser) {
-        return withContext(context -> getRelayConfigurationWithResponse(communicationUser, context));
+        return withContext(context -> getRelayConfigurationWithResponse(communicationUser, null, context));
     }
 
     /**
      * Creates a new CommunicationRelayConfiguration with response.
      *
+     * @param routeType The specified CommunicationRelayConfigurationRequestRouteType for the relay request
      * @return The created Communication Relay Configuration.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse() {
-        return withContext(context -> getRelayConfigurationWithResponse(null, context));
+    public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationRelayConfigurationRequestRouteType routeType) {
+        return withContext(context -> getRelayConfigurationWithResponse(null, routeType, context));
+    }
+
+    /**
+     * Creates a new CommunicationRelayConfiguration with response.
+     *
+     * @param communicationUser The CommunicationUserIdentifier for whom to issue a token
+     * @param routeType The specified CommunicationRelayConfigurationRequestRouteType for the relay request
+     * @return The created Communication Relay Configuration.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationUserIdentifier communicationUser, CommunicationRelayConfigurationRequestRouteType routeType) {
```

Comment:
We have 4 methods for getRelayConfiguration operation that returns `WithResponse`. Ideally, for an operation we would have just 1 `WithResponse` method that represents the max overload.

Reply:
Oh I didn't know that, I though we needed all the overloads for WithResponse.